specification/datadog/resource-manager/Microsoft.Datadog/preview/2020-02-01-preview/examples/MarketplaceAgreements_List.json

```diff
@@ -0,0 +1,45 @@
+{
+  "parameters": {
+    "api-version": "2020-02-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.MarketplaceOrdering/agreements/id1",
+            "name": "planid1",
+            "type": "Microsoft.MarketplaceOrdering/offertypes",
+            "properties": {
+              "publisher": "pubid1",
+              "product": "offid1",
+              "plan": "planid1",
+              "licenseTextLink": "test.licenseLink1",
+              "privacyPolicyLink": "test.privacyPolicyLink1",
+              "retrieveDatetime": "2017-08-15T11:33:07.12132Z",
+              "signature": "ASDFSDAFWEFASDGWERLWER",
+              "accepted": false
+            }
+          },
+          {
+            "id": "/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.MarketplaceOrdering/agreements/id2",
```

Comment:
if there are multiple items being returned in a list, there should be a corresponding put for each as well, since if there are 2 agreements there should be two accepts - else there should only be 1 agreements resource (default) returned.

Reply:
Let me explain our API use cases. Datadog is a Marketplace resource. Before the user register to use Datadog RP, some marketplace terms are needed to be accepted. For example, the general marketplace term and Datadog service specific terms. The requirement is to have 2 subscription level APIs. One API is to list all these terms. One API is to accept all these terms. There are no real resource will be created. All these APIs are working like a proxy. It will always show all the terms or accept all the terms as one big term(default one).

Reply:
if you want a single api to accept multiple terms, you should model the terms as an array within a singleton agreements resource called default - then the get, put and list (returning 1 agreement resource) will be consistent. the way you have it modeled today I feel is not consistent where the customer can get multiple agreements resources, but then accepts both by accepting a 'default' one.

Reply:
The default resource is not a real resource. It is only used for accepting the terms. The POST action will be better. But as the RPaaS currently didn't support the POST action in the subscription level. So PUT is used here temporarily. See: https://github.com/Azure/azure-rest-api-specs-pr/pull/2309#discussion_r516467148. The long term solution will be using POST once RPaaS is supporting. The terms will include at least 2 terms(marketplace term and Datadog term). They will always be showing together.
This is discussed in the PR and meeting for https://github.com/Azure/azure-rest-api-specs-pr/pull/2309. Can we go with this now as RPaaS PR has been merged? We will change later once RPaaS supports POST action for subscription level.

Reply:
I still don't fully agree with this model - @ravbhatnagar to take a look, since he seemed to have the most context on the other PR